src/v/v8_engine/script.h

```diff
@@ -55,9 +63,32 @@ class script {
     /// Compile js code and get function for run in future.
     ///
     /// \param function name for runnig in future
-    /// \param buffer with js code for compile
-    void init(std::string_view name, ss::temporary_buffer<char> js_code);
-    void run(ss::temporary_buffer<char>& data);
+    /// \param buffer with js code for compile. TODO: think about
+    /// temporary_buffer. Maybe better use iobuf and use non-seastar memory for
+    /// data in v8 script. \param executor for compile script
+    template<typename Executor>
+    ss::future<> init(
+      ss::sstring name,
+      ss::temporary_buffer<char> js_code,
+      Executor& executor) {
+        compile_task task(*this, std::move(js_code));
+        return add_future_handlers(
+                 executor.submit(
+                   std::forward<compile_task>(task), _first_run_timeout_ms))
```

Comment:
std::move(task) not std::forward, as it is not a universal ref. Use std::forward when the parameter is a double ref (&&) to a templated type

Reply:
done